src/device-state.ts

```diff
@@ -468,7 +467,7 @@ function emitAsync<T extends keyof DeviceStateEvents>(
 		: Array<DeviceStateEvents[T]>
 ) {
 	if (_.isArray(args)) {
-		return setImmediate(() => events.emit(ev as any, ...args));
+		return setImmediate(() => events.emit(ev as any, ...(args as any[])));
```

Comment:
Why did this become necessary?

Reply:
@Page it was popping up an error whenever I opened the file, this was for OCD purposes (it doesn't result in failed lint checks though)

Reply:
What was the linter/transpiler yelling exactly ? Seems odd that it's complaining now.

Reply:
@20k-ultra 
```
Type 'DeviceStateEvents[T] extends (...args: any) => void ? Parameters<DeviceStateEvents[T]> : DeviceStateEvents[T][]' must have a '[Symbol.iterator]()' method that returns an iterator. ts(2488)
```

It's one of those red squiggly line errors that don't fail lint checks or transpilation

Reply:
Try to see what you can find out when searching 'must have a '[Symbol.iterator]()' method that returns an iterator.". Maybe there's something we should do.

Reply:
The function types are pretty strange, my guess is that they are left from an older version of typescript, maybe you xcan try something like 

```
// We keep compatibility with the StrictEventEmitter types
// from the outside, but within this function all hells
// breaks loose due to the liberal any casting
function emitAsync<T extends keyof DeviceStateEvents>(
	ev: T,
	...args: DeviceStateEvents[T] extends (...args: infer TArgs) => void
		? TArgs
		: Array<DeviceStateEvents[T]>
) {
	if (_.isArray(args)) {
		return setImmediate(() => events.emit(ev as any, ...args));
	} else {
		return setImmediate(() => events.emit(ev as any, args));
	}
}
```

Reply:
I'm not 100% sure, but I think the `isArray` check is not necessary, since `args` should always be an array given the `...`, even if the function is called with a single argument

Reply:
@pipex Thanks! This works, and also you're right, the isArray check isn't needed.